gwcs/region.py

```diff
@@ -162,6 +162,11 @@ def scan(self, data):
         scline = self._scan_line_range[-1]
         while y <= scline:
             AET = self.update_AET(y, AET)
+
+            if self._bbox[2] <= 0:
+                y += 1
+                continue
+
```

Comment:
Is this is the right place for this check? A polygon has a defined bounding box and it does not change with the scan line. I think this check should be moved to the beginning of the scan method and it should simply return `data`.
```
if self._bbox[2] <= 0:
    return data
```

Reply:
You are so right. I missed this.